app/models/storage_location.rb

```diff
@@ -145,7 +148,8 @@ def increase_inventory(itemizable_array)
       # Locate the storage box for the item, or create a new storage box for it
       inventory_item = inventory_items.find_or_create_by!(item_id: item_hash[:item_id])
       # Increase the quantity-on-record for that item
-      new_quantity = inventory_item.quantity + item_hash[:quantity].to_i
+      new_quantity = item_hash[:quantity].to_i
+      new_quantity += inventory_item.quantity unless replace_inventory
       inventory_item.update!(quantity: new_quantity)
       # Record in the log that this has occurred
       log[item_hash[:item_id]] = "+#{item_hash[:quantity]}"
```

Comment:
Note that this log is now incorrect - we didn't add this quantity, we added (new_quantity - old_quantity).

I think I'd prefer being more explicit here - rather than setting the inventory to the given value, we should calculate the difference and do an `increase_inventory` or `decrease_inventory` to the value we've calculated. I think there would be some edge cases that would bite us if we don't do it this way.